### PR TITLE
limit total numbers of spaces

### PR DESCRIPTION
**Pull request by knightXun**
We must limit the number of spaces, that's important.

### CODE REVIEW

**Comment by Shylock-Hg** [comment]
Could treat negative as infinity limit.

File: src/interface/meta.thrift [hunk 1]
```diff
@@ -21,6 +21,7 @@ enum ErrorCode {
     E_LEADER_CHANGED   = -11,
 
     // Operation Failure
+    E_TOO_MANY_SPACE   = -20,
```
Comment (yixinglu) [inline src/interface/meta.thrift]: `E_TOO_MANY_SPACES `

Comment (knightXun) [reply src/interface/meta.thrift]: thx

File: src/meta/processors/partsMan/CreateSpaceProcessor.cpp [hunk 2]
```diff
@@ -9,6 +9,8 @@
 
 DEFINE_int32(default_parts_num, 100, "The default number of parts when a space is created");
 DEFINE_int32(default_replica_factor, 1, "The default replica factor when a space is created");
+DEFINE_int32(default_space_sum, 100, "The default number of space can be created");
+DEFINE_int32(default_max_replica_factor, 1, "The default max replica factor that a space can take");
```
Comment (dutor) [inline src/meta/processors/partsMan/CreateSpaceProcessor.cpp]: Please revise the naming of these two flags. _**default**_ is the value we take when user doesn't provide. You could infer from the fact how `default_parts_num` are used.

Comment (knightXun) [reply src/meta/processors/partsMan/CreateSpaceProcessor.cpp]: good point, I changed these two flag names

File: src/meta/processors/partsMan/CreateSpaceProcessor.cpp [hunk 3]
```diff
@@ -119,6 +137,22 @@ CreateSpaceProcessor::pickHosts(PartitionID partId,
...
+int32_t CreateSpaceProcessor::getSpaceSum() {
+    folly::SharedMutex::ReadHolder rHolder(LockUtils::spaceLock());
+    int32_t count = 0; 
+    auto prefix = MetaServiceUtils::spacePrefix();
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto ret = kvstore_->prefix(kDefaultSpaceId, kDefaultPartId, prefix, &iter);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        return -1;
+    }
+    std::vector<cpp2::IdName> spaces;
```
Comment (yixinglu) [inline src/meta/processors/partsMan/CreateSpaceProcessor.cpp:149]: It seems an unuseful variables: `spaces`

Comment (knightXun) [reply src/meta/processors/partsMan/CreateSpaceProcessor.cpp:149]: Good point, I will carefully rewrite this pr

**Pull request closed without merging**